### DISCUSSION TITLE
Microsoft + Github

### DISCUSSION THREAD

**Discussion by sbvorpagel** [frontendbr/forum#1109] [Discussão]
Pessoal, como vocês estão vendo essa aquisição?
Será que isso pode afetar a questão forte de opensource que sempre teve no github? Comunidades? etc?

**Comment by felipefialho** [comment]
Dei minha opinião no [Twitter](https://twitter.com/LFeh/status/1003451819060809729) e vou replicar aqui 


>Sobre essa aquisição do Github pela Microsoft.
>
>Devíamos ficar felizes por uma empresa tão importante na evolução do desenvolvimento de software >conseguir lucro.
>
>O Github fez tudo certinho:
>
>- Nasceu
>- Amadureceu
>- Morr... foi vendido
>
>É quase um amigo próximo ganhando na loteria.
>
>A Microsoft errou muito no passado, mas nos últimos anos melhorou. Inclusive lançou projetos open->source importantes, entre eles o VSCode.
>
>Então, calma gente, não é como se o Github estivesse fazendo um pacto com o demo ... ou talvez seja 😈 

De verdade, não da pra afirmar muita coisa nesse momento, mas acho dificil que a gente sinta grandes diferenças.

Importante lembrar que a Microsoft comprou o LinkedIn faz pouco tempo, e houveram melhorias no serviço.

**Comment by didiraja** [comment]
Eu fico bolado não com isso do open source mas com decisões ruins da MS, vide o Skype que foi definhando aos poucos. Além de todo o lance de dados né, Github era um grupo independente não uma empresa gigante.

**Comment by felipefialho** [comment]
@didiraja Acho que o que matou o Skype foi o aparecimento de mensageiros mais eficazes, como o appear.in

**Comment by fredmosc** [comment]
Acredito que é uma aposta da MS é retomar o controle na parte de
servidores, eles estão com ótimas iniciativas no open-source visando
fortalecer a Azure.
Lógico que a parte do lucro que o Github tem, influencia na compra. Mas to
vendo muita gente desesperada, migrando para o Gitlab sem nem saber o
posicionamento da empresa.
Vamos ver no que dá

Em seg, 4 de jun de 2018 às 12:25, Felipe Fialho <notifications@github.com>
escreveu:

> Dei minha opinião no Twitter
> <https://twitter.com/LFeh/status/1003451819060809729> e vou replicar aqui
>
> Sobre essa aquisição do Github pela Microsoft.
>
> Devíamos ficar felizes por uma empresa tão importante na evolução do desenvolvimento de software conseguir lucro.
>
> O Github fez tudo certinho:
>
> - Nasceu
> - Amadureceu
> - Morr... foi vendido
>
> É quase um amigo próximo ganhando na loteria.
>
> A Microsoft errou muito no passado, mas nos últimos anos melhorou. Inclusive lançou projetos open-source importantes, entre eles o VSCode.
>
> Então, calma gente, não é como se o Github estivesse fazendo um pacto com o demo ... ou talvez seja 😈
>
> De verdade, não da pra afirmar muita coisa nesse momento, mas acho dificil
> que a gente sinta grandes diferenças.
>
> Importante lembrar que a Microsoft comprou o LinkedIn faz pouco tempo, e
> houveram melhorias importantes no serviço.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/1109#issuecomment-394395053>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJB79SyqYAkcpI9fjYF91V6WEwYXu68Xks5t5VFbgaJpZM4UZPxb>
> .
>
-- 
Att,

FM
<https://about.me/fredmosc?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=edit_panel&utm_content=thumb>
Frederick Moschkowich
about.me/fredmosc
<https://about.me/fredmosc?promo=email_sig&utm_source=product&utm_medium=email_sig&utm_campaign=edit_panel&utm_content=thumb>

**Comment by fabriciofmsilva** [comment]
Não adianta tentar fazer um trabalho de futurologia aqui. Minha opinião é: acompanhar a evolução (lembrando que evolução pode melhorar ou piorar), se melhorar continuarei usando, se piorar procuro uma alternativa. Certa vez ouvi uma frase: "Se é bom ou se é mau, só o futuro dirá" - Rubem Alves. Sigo pensando assim.

**Comment by fabiorocha** [comment]
[Disclaimer, trabalho na MSFT]

Como o pessoal falou aí emcima, é muito cedo pra tirar qualquer conclusão definitiva. Mas vamos aos fatos que estão alimentando tanta **especulação** sobre o futuro.

De um lado, o pessoal falando sobre a Microsoft, mas basicamente sobre a Microsoft de 5, 10, 15, 20 anos atrás: a Microsoft fechada, e que queria garantir o espaço do Windows a todo e qualquer custo; a Microsoft que comprou a Nokia e o Skype e os deixou cair de qualidade/entrar em extinção.

De outro, a Microsoft de 5 anos pra cá: aberta, que tem investido pesadamente em Open Source, inclusive abrindo o código de cada vez mais projetos -- e tem MUITO time aqui trabalhando pra virar OSS o mais breve possível. Tem uma estragégia por trás de todo esse investimento? Claro que tem. A Microsoft ainda é uma empresa como qualquer outra, mas o que eu vejo é que a cultura é muito mais altruísta e o modelo de negócios tá mais atrelado ao Enterprise do que ao usuário final.

Milhares de pessoas aqui na Microsoft usam o GitHub como qualquer pessoa fora da Microsoft. Se o pessoal não gerenciar bem, vão ouvir o feedback alto e claro de nós mesmos antes que qualquer outra coisa. Então eu to otimista que vão tratar o GitHub melhor que qualquer outro projeto aqui dentro.

**Comment by eduardomarino** [comment]
Espero que não prejudique o desenvolvimento do Atom Editor.

**Comment by ninetails** [comment]
bom, o LinkedIn não chegou a mudar drasticamente nem teve algo que pessoalmente tenha me afetado, e gostei muito desta posição pós aquisição. Espero que com o Github seja o mesmo.

Claro, espero MUITO também que não comecem a empurrar soluções Azure Cloud só por usar Github nem que mudem minha subscription de 7 obamis unlimited private repos :(

**Comment by sbvorpagel** [comment]
Meu medo é a questão de comunidade e inflar muito de recursos voltados ao ferramental da microsoft.

**Comment by caioincau** [comment]
Estou super empolgado, quem sabe o Electron se torne algo bom e menos memmory hogger.

**Comment by ortizvinicius** [comment]
Microsoft nos deu o VSCode e pediu o GitHub em troca

**Comment by klarkc** [comment]
Só sei q o GitLab já tá [sofrendo as consequências](https://twitter.com/gitlabstatus) rs.

**Comment by luizbills** [comment]
Daqui uns dias vai ter um botãozinho "deploy to azure" em nossos repos

**Comment by klarkc** [comment]
Pra mim tem muito mais a ver com o LinkedIn do que com o Azure ou qualquer outro serviço. Imagina o poder que a Microsoft agora tem, com dados de ambas as plataformas. Prevejo uma plataforma revolucionária de contratação em breve.

**Comment by felipefialho** [comment]
@klarkc Mano... você levantou um ponto que não tinha pensado heim...

**Comment by tarciozemel** [comment]
Bitbucket já começou a auxiliar a comunidade nesse momento difícil... :laughing: 

![captura de tela de 2018-06-05 09-32-27](https://user-images.githubusercontent.com/133777/40976149-86a7e79e-68a3-11e8-8db7-f9f3d6118ba6.png)

**Comment by klarkc** [comment]
https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT

**Comment by caioincau** [comment]
Tempestade no copo de água, MS mudou muito, plus, serão duas empresas separadas, o Linkedin mesmo, teve melhorias após a  compra.

Microsoft pode ser divido entre durante e depois da saída do Steve Ballmer.

**Comment by Rodrigo54** [comment]
Eu acho que eles compraram o github pq vão abrir o código fonte do windows, maaaas é só especulação :joy: :joy: :joy:

**Comment by ninetails** [comment]
@Rodrigo54 :

> [Microsoft’s whole Windows 10 platform is now built on open-source Progressive Web App technology](https://medium.com/@ow/microsoft-acquiring-github-is-a-good-thing-heres-why-6a6a57eb83ac)

btw:

> [Each of the three could receive about 12.3 million Microsoft shares, assuming they control equal stakes in GitHub. A 0.16 percent holding would give each of them about 10 times more shares than Chief Executive Officer Satya Nadella, and roughly 14 times more than President Brad Smith. Among insiders, only co-founder Bill Gates’s 1.34 percent holding would exceed theirs, according to data compiled by Bloomberg.](https://www.bloomberg.com/news/articles/2018-06-04/github-billionaires-will-own-more-microsoft-stock-than-its-ceo)

**Comment by Murillofilho86** [comment]
E ai, quem vcs iriam preferir, Microsoft ou Google??? 
[https://www.cnbc.com/2018/06/05/github-interest-from-google-and-others-revenue-about-300-million.html](url)

**Comment by renanbastos93** [comment]
Eu só sei que vejo muitos usando VS Code e condenando a microsoft.
Eu estou bem tranquilo, e sei que a Microsoft tem mudado muito, até participei de projetos para  fomentar tecnologia OpenSource, e também treinando novos profissionais. Não vejo isso como o fim do mundo, e sim como novas oportunidades que vão surgir para a comunidade.

**Comment by klarkc** [comment]
hue [esse repositório estava em #1 no trendings, e foi censurado](https://github.com/upend/IF_MS_BUYS_GITHUB_IMMA_OUT#breaking-news-microsoft-github-just-censored-this-project-from-its-trending-page) segundo eles

**Comment by rsmelo92** [comment]
Nat Friedman o novo CEO do github
https://natfriedman.github.io/hello/

**Comment by ninetails** [comment]
> Nat Friedman o novo CEO do github
> https://natfriedman.github.io/hello/

Bom, pelo menos curti a postura apesar de um CEO indicado provavelmente. Talvez fique mesmo estilo LinkedIn de aquisição.

**Comment by vsppedro** [comment]
Quais foram os projetos open source de peso que largaram o GitHub até agora?

Eu ainda não vejo motivos para não dar uma chance a MS.

E creio que é muito cedo para falar.

**Comment by kvnol** [comment]
:top:

**Comment by JonatasRicardo** [comment]
Notei que notei que meu github foi loggado em um pc em Belo Horizonte (sou do Rio de Janeiro).
Será que alguém invadiu? Fiquei preocupado.
Coincidentemente a Microsoft tem um escritório em BH (https://www.microsoft.com/pt-br/about/onde-estamos.aspx).

Aconteceu com mais alguém? Para ver basta ir em https://github.com/settings/security e dar uma olhada em "Sessions".

**Comment by felipefialho** [comment]
@JonatasRicardo O meu ta suave, hahaha

**Comment by Marlysson** [comment]
@JonatasRicardo Aqui tbm de boa..

Sobre a discussão, como muitos já falaram, a Microsoft de uns tempos pra cá mudou bastante, contribuiu bastante para a comunidade com ótimas ferramentas que auxiliam os desenvolvedores.

Não acredito que ela vá acabar com o GitHub , no máximo fazer algumas integrações com os serviços dela... Mas acabar com um serviço desses acho que seria um grande tiro no pé.

Aqui encontrei um artigo com itens do que a Microsoft pode fazer com o GitHub:

[6 Things Microsoft Plans to Do With GitHub](https://www.pcmag.com/feature/361545/6-things-microsoft-plans-to-do-with-github)

**Comment by vsppedro** [comment]
@JonatasRicardo O meu ta de boa tbm. Já teria mudado a senha umas três vezes e ainda atualizado o 'two factory authentication'.

**Comment by JonatasRicardo** [comment]
Sobre a "invasão" na minha conta:  o github reconhece meu celular como sendo de BH :smile:

**Comment by mahenrique94** [comment]
Microsoft está melhorando cada vez mais, não podemos ficar presos no passado, ela já lançou vários projetos e vem evoluindo. O principal de mudar seu conceito e politico deve ter sido feito.

- TypeScript
- VSCode
- Visual Studio para MAC
- SQL Server para Linux

Entre outros, devemos ficar felizes e não tristes com o acontecido.

**Comment by alexsandro-xpt** [comment]
@JonatasRicardo acha mesmo a MS iria acessar seus repositórios privados da sua conta usando Usuário e Senha sendo que ela já tem todo o banco de dados?!

Não se preocupe.

**Comment by JonatasRicardo** [comment]
@alexsandro-xpt tô de boa rs

**Comment by dougaraujos** [comment]
Pra entrar no clima 
https://github.com/martenbjork/github-xp

**Comment by MarcosEllys** [comment]
Vou compartilhar alguns pontos aqui que já indaguei com colegas:

1 - Uma coisa é a M$ tocar projetos Open como Type Script ou VS Code (Projetos).
2 - Outra coisa é ela comprar a maior plataforma SaaS de repostórios.

Todo mundo fala sobre a M$ e o quanto ela é uma empresa friendly com Open Source, mas o Github passa longe de ser uma plataforma Open Source, é um **produto** SaaS e o que eu espero pro futuro é uma separação mais clara sobre o free e o premium, como já é assim no Linkedin.

PS: Tente ver o Github como uma plataforma SaaS antes de criticar, a pegada dos repositórios Open Source, Pull Request vai continuar, mais como produto mesmo podemos ver algumas mudanças (talvez boas como Pipeline like Bitbucket) em um futuro breve

**Comment by adrianoavelino** [comment]
só complementando que agora será possível repositórios privados e free, mas com restrição na quantidade contribuidores: [https://blog.github.com/2019-01-07-new-year-new-github/](url)

**Comment by felipefialho** [comment]
Muito bem colocado @adrianoavelino!

**Comment by giovannism20** [comment]
@adrianoavelino Testei e encontrei algumas outras objeções (apenas complementando): 

> GitHub Pages
> GitHub Pages is designed to host your personal, organization, or project pages from a GitHub repository.
> 
> Upgrade to GitHub Pro or make this repository public to enable Pages.

> Wikis 
> GitHub Wikis is a simple way to let others contribute content. Any GitHub user can create and edit pages to use for documentation, examples, support, or anything you wish.
> 
> Upgrade to GitHub Pro or make this repository public to enable Wikis.

**Comment by giovannism20** [comment]
@mahenrique94 Tirando o track do VSCode, tá de boa. rsrs

**Comment by mksanti** [comment]
@GiovanniSM20 isso mesmo.

![image](https://user-images.githubusercontent.com/5166571/50899554-29bfea00-13f1-11e9-8520-57e1c5dc9e02.png)

Os detalhes de cada plano pode ser encontrados na [pagina de preço](https://github.com/pricing) do GH

Mas esse lance pode ser contornado pela infinidade de possibilidades de hospedagem de conteúdo estático, como Netlify, Surge etc...

**Comment by WandersonAlves** [comment]
Pergunta sobre o pricing do github:

Caso eu crie uma page, onde nessa page vão ter N repos (privados ou não) e N colaboradores, eu poderia usar o plano 'Pro' nessa page e sair dando invite pros usuários codarem nos repos?

EDIT: essa page seria uma organização no caso kkkkkkk

**Comment by giovannism20** [comment]
Sim, você estará colocando colaboradores dentro de sua organização @WandersonAlves . Ou, eles podem apenas mandarem PR's para participar (caso sejam de fora).

p.s: E sim, entendi que essa ideia seria tipo uma forma de burlar o sistema.

**Comment by giovannism20** [comment]
Queria comentar também que acabei de entrar na página do meu perfil e o GitHub adicionou um status (não como um what's app) mas algo como temos no Slack.

**Comment by NicholasKuchiniski** [comment]
Está muito bom! Mudanças positivas por enquanto. Espero que deem mais opções para customizar o perfil, a criação de comunidades (não organizações) e etc.

**Comment by ninetails** [comment]
sobre o Status, o que mais curti foi que vc pode dar seu feedback logo do lado da funcionalidade, e não ter que buscar a web toda por uma issue, um fórum etc

copiando o que mandei: achei meio confuso sobre esta funcionalidade pois não sei se o status é igual o Slack (direcionado pra qualquer um que vê seu perfil) ou se é algo mais pra comunicar coworkers

**Comment by giovannism20** [comment]
Também não sei @ninetails até por conta da opção 'busy'. Achei parecido com a do escola porquê no mesmo, temos essa opção e assim podemos notificar a todos a respeito do que está sendo feito e sobre dar feedback, também achei sensacional (só resta saber se não é inútil o mesmo...)

**Comment by giovannism20** [comment]
Além do que estava vendo agora (dentro da issue|só percebi agora) eles andaram mudando algumas coisas mesmo que mínimas no layout.

https://imgur.com/a/3gJyITX

**Comment by WandersonAlves** [comment]
> Sim, você estará colocando colaboradores dentro de sua organização @WandersonAlves . Ou, eles podem apenas mandarem PR's para participar (caso sejam de fora).
> 
> p.s: E sim, entendi que essa ideia seria tipo uma forma de burlar o sistema.

Não dá pra fazer como pensei kkk @GiovanniSM20 

![image](https://user-images.githubusercontent.com/9141449/50993320-2400fc00-14f8-11e9-92bd-ecf7421d558a.png)

**Comment by giovannism20** [comment]
Acho que uma coisa legal que a Microsoft podia fazer dentro do GitHub é uma lugar para sugestões dentro da plataforma.
